lib/puppet/ssl/host.rb

```diff
@@ -371,27 +378,61 @@ def state
 
   private
 
+  # Ensures that CRL is either on disk, or that CRL checking has been disabled
+  def ensure_crl!
+    if crl_usage && !Puppet::FileSystem.exist?(crl_path)
+      # The CertificateRevocationList indirector will attempt to download the
+      # CRL from the CA if it does not exist on disk.
+      if !CertificateRevocationList.indirection.find(name)
+        Puppet.debug _("Certificate revocation checking is enabled but a CRL cannot be found; CRL checking will not be performed.")
+        @crl_usage = false
+      end
+    end
+  end
+
+  # @param path [String] Path to CRL Chain
+  # @return [Array<OpenSSL::X509::CRL>] CRLs from chain
+  # @raise [Errno::ENOENT] if file does not exist
+  # @raise [Puppet::Error<OpenSSL::X509::CRLError>] if the CRL chain is malformed
+  def load_crls(path)
+    ending = "-----END X509 CRL-----\n"
+    crls_pems = Puppet::FileSystem.read(path)
+    crls_pems.split(ending).map do |crl|
+      begin
+        crl += ending
+        OpenSSL::X509::CRL.new(crl)
+      rescue OpenSSL::X509::CRLError => e
+        raise Puppet::Error.new(
```

Comment:
Prior to this the `Puppet::SSL::Base` class that the `CertificateRevocationList` indirection subclasses passed any string given to it directly into `OpenSSL::X509::CRL.new` so I assume this changes the interface from raising `OpenSSL::X509::CRLError` to `Puppet::Error`, I assumed since this is targeting a major and we can get a better error message that's fine?